Data Masking included in Message Analysis.md (#360)
hellohaptik/docs_haptik: docs/bot-analytics/using-message-analysis.md

#### Predefined Masked Entities on the platform

Few entities are considered as **personal** at the platform level. Once they are detected in conversations, they are automatically masked. This list is not editable and is defined at the platform level. Currently, only one entity is under this list -

- phone_number

#### Predefined Regex Patterns Entities on the platform

Few entities are considered as **personal** using **Regex Patterns**. Once they are detected in conversations, they are automatically masked. This list is not editable and is defined at a Platform level. Currently, the below regex patterns are used for this -

- **r_email** - \b[A-Za-z0-9\.\+_-]+@[A-Za-z0-9\._-]+\.[a-zA-Z]+\b
- **r_numbers** - \b\d{6,16}\b
- **r_pancard** - \b[A-Za-z]{5}\d{4}[A-Za-z]\b

### Enable Data Masking

Data masking will be enabled by default for all new partners being created. If you wish to switch off the Data Masking (Data Encryption) flag, you can do the same from the Partners List under Admin Tools as shown below

![PersonalEntities2](assets/DataMasking2.png)

We do not recommend disabling Data Masking. If you do, then please explicitly inform the same among all stakeholders.

> ### Can we see the unmasked/decrypted message?

Author: ```suggestion
```

> Note: When you apply multiple filters, it is possible for the filters to cancel out each other or overlap each other. In such cases you might not see any data or see unexpected data. In such cases it is useful to use this button to clear all filters.

![Clear Filters](assets/message-analysis-clear-filters.png)

## Data Masking

Personal information in a conversation needs to be protected when a conversation is marked as complete. This includes User Says, Bot Says, Agent Messages, Media, etc. We mask or encrypt this personal information in order to protect it. Data masking is the practice of intentionally randomizing anonymizing specific data/information to secure it. 

We mask the **original_text** not **entity_value**. In a few scenarios, the original_text and entity value are different. The current masking rules consider only the original_text. In the below image, it is shown how **original_text** and **entity_value** can be different -

![PersonalEntities](assets/DataMasking3.png)

> Data on **Message Analysis** screen will show up in the masked format. If you download the **Message Analysis report**, then that will also show up with the masked version. However, chat links will show up in their original form. For easier bot maintenance, the masked value will also be tagged with what kind of information it is. Below image shows how a masked entity value will look like

Author: ```suggestion
Data on **Message Analysis** screen will show up in the masked format. If you download the **Message Analysis report**, then that will also show up with the masked version. However, chat links will show up in their original form. 

The masked data is shown as 6 asterisks in a message as shown below -
```

> Note: When you apply multiple filters, it is possible for the filters to cancel out each other or overlap each other. In such cases you might not see any data or see unexpected data. In such cases it is useful to use this button to clear all filters.

![Clear Filters](assets/message-analysis-clear-filters.png)

## Data Masking

Personal information in a conversation needs to be protected when a conversation is marked as complete. This includes User Says, Bot Says, Agent Messages, Media, etc. We mask or encrypt this personal information in order to protect it. Data masking is the practice of intentionally randomizing anonymizing specific data/information to secure it. 

We mask the **original_text** not **entity_value**. In a few scenarios, the original_text and entity value are different. The current masking rules consider only the original_text. In the below image, it is shown how **original_text** and **entity_value** can be different -

![PersonalEntities](assets/DataMasking3.png)

> Data on **Message Analysis** screen will show up in the masked format. If you download the **Message Analysis report**, then that will also show up with the masked version. However, chat links will show up in their original form. 

Reviewer: However, **the messages on** chat links will show up in their original form.

**chat links** should have a reference to chat link documentation page.

Author: There is no such docs currently. Once I add, will add the link here too.

The masked data is shown as 6 asterisks in a message as shown below -

![PersonalEntities](assets/DataMasking4.png)

### What all gets masked? 

> Not everthing needs to be protected but sensitive data like phone-number, email-id, account number, date-of-birth, etc. needs to be protected. On the platform, these values are inserted as **entities**. So we essentially need to mask these sensitive entities.

Author: ```suggestion
Not everything needs to be protected but sensitive data like phone-number, email-id, account number, date-of-birth, etc. needs to be protected. On the platform, these values are inserted as **entities**. So we essentially need to mask these sensitive entities.
```

> > Nothing else gets masked except **entities**.

Reviewer: Not sure if this phrase is valid.
Regex patterns will tag phrases which may not correspond to an entity.

Author: Is it? I thought only entities are masked. Thats what we discussed yesterday right?

Few entities are considered as **personal** at the platform level. Once they are detected in conversations, they are automatically masked. This list is not editable and is defined at the platform level. Currently, only one entity is under this list -

- phone_number

#### Predefined Regex Patterns Entities on the platform

Few entities are considered as **personal** using **Regex Patterns**. Once they are detected in conversations, they are automatically masked. This list is not editable and is defined at a Platform level. Currently, the below regex patterns are used for this -

- **r_email** - \b[A-Za-z0-9\.\+_-]+@[A-Za-z0-9\._-]+\.[a-zA-Z]+\b
- **r_numbers** - \b\d{6,16}\b
- **r_pancard** - \b[A-Za-z]{5}\d{4}[A-Za-z]\b

### Enable Data Masking

Data masking will be enabled by default for all new partners being created. If you wish to switch off the Data Masking (Data Encryption) flag, you can do the same from the Partners List under Admin Tools as shown below

![PersonalEntities2](assets/DataMasking2.png)

We do not recommend disabling Data Masking. If you do, then please explicitly inform the same among all stakeholders.

### Can we see the unmasked/decrypted message?

> Even though it is technically possible to view the decrypted message, we do not allow the same anywhere. If absolutely required, then please get an approval from Haptik's Infosec team before using it in any application.

Author: ```suggestion
```
